test/suites/integration/instance_tag_test.go

```diff
@@ -0,0 +1,144 @@
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+		}}
+
+		provisionerOpts = test.ProvisionerOptions{
+			Requirements: []v1.NodeSelectorRequirement{
+				{
+					Key:      v1alpha5.LabelCapacityType,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{v1alpha1.CapacityTypeOnDemand},
+				},
+				{
+					Key:      v1.LabelOSStable,
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{string(v1.Linux)},
+				},
+				{
+					Key:      "karpenter.k8s.aws/instance-hypervisor",
+					Operator: v1.NodeSelectorOpIn,
+					Values:   []string{"nitro"},
+				},
+			},
+			Limits: v1.ResourceList{},
+		}
+	})
+
+	It("should overwrite default Name tags", func() {
+		nodeTemplate := awstest.AWSNodeTemplate(nodeTemplateOpts)
+		provisioner := test.Provisioner(provisionerOpts, test.ProvisionerOptions{ProviderRef: &v1alpha5.MachineTemplateRef{
+			Name: nodeTemplate.Name,
+		}})
+		pod := test.Pod()
+
+		env.ExpectCreated(provisioner, nodeTemplate, pod)
+		env.EventuallyExpectCreatedMachineCount("==", 1)
+		env.EventuallyExpectCreatedNodeCount("==", 1)
+		node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
+		nodeName := client.ObjectKeyFromObject(node)
+
+		Eventually(func(g Gomega) {
+			node = &v1.Node{}
+			lo.Must0(env.Client.Get(env.Context, nodeName, node))
+			g.Expect(node.Annotations["karpenter.sh/instance-tagged"]).To(Equal("true"))
+		}, 30*time.Second)
+
+		resourceID := lo.Must(getResourceID(node.Spec.ProviderID))
+		tags := getResourceTags(resourceID)
+		Expect(tags["Name"]).To(Equal(node.Name))
+	})
+
+	It("shouldn't overwrite custom Name tags", func() {
+		nodeTemplate := awstest.AWSNodeTemplate(nodeTemplateOpts, v1alpha1.AWSNodeTemplateSpec{AWS: v1alpha1.AWS{
+			Tags: map[string]string{
+				"Name": "custom-name",
+			},
+		}})
+		provisioner := test.Provisioner(provisionerOpts, test.ProvisionerOptions{ProviderRef: &v1alpha5.MachineTemplateRef{
+			Name: nodeTemplate.Name,
+		}})
+		pod := test.Pod()
+
+		env.ExpectCreated(provisioner, nodeTemplate, pod)
+		env.EventuallyExpectCreatedMachineCount("==", 1)
+		env.EventuallyExpectCreatedNodeCount("==", 1)
+		node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
+		nodeName := client.ObjectKeyFromObject(node)
+
+		Eventually(func(g Gomega) {
+			node = &v1.Node{}
+			lo.Must0(env.Client.Get(env.Context, nodeName, node))
+			g.Expect(node.Annotations["karpenter.sh/instance-tagged"]).To(Equal("true"))
+		}, 30*time.Second)
+
+		resourceID := lo.Must(getResourceID(node.Spec.ProviderID))
+		tags := getResourceTags(resourceID)
+		Expect(tags["Name"]).To(Equal("custom-name"))
+	})
+})
+
+func getResourceID(providerID string) (string, error) {
+	id := path.Base(providerID)
+	if id == "." || id == "/" {
+		return "", fmt.Errorf("failed to extract resourceID from providerID")
+	}
+	return id, nil
+}
+
+func getResourceTags(resourceID string) map[string]string {
```

Comment:
Checkout the `test/pkg/environment/aws/expectations.go` file. I think there may be some functions in there (like `env.GetInstance`) that should help you be able to get some of this information without having to build-out your own functions